src/rmath.jl

```diff
@@ -6,6 +6,8 @@ import Rmath: libRmath
 
 ### import macro
 
+const FloatOrInt = Union{Float16,Float32,Float64,Int}
```

Comment:
Any reason not to accept smaller integer types? Maybe we could even allow any `Real`, and let the conversion throw in case of overflow?

Reply:
I did not know where to stop so I just started with the types that would fix the linked issue in Distributions :smile: I was wondering about smaller integer types as well. I think `Real` would break stuff and lead to method ambiguities since for some of these functions we already define Julia implementations that accept `Real` arguments in StatsFuns (they are used e.g. for `Dual` numbers).

Reply:
Hmm, then why not use the Julia implementations when they exist? Is it faster to call Rmath even if that involves conversions?

Reply:
This is the motivation for https://github.com/JuliaStats/StatsFuns.jl/pull/113 which I did not want to touch or duplicate in this PR. I haven't done any benchmarks but in my opinion performance should not be the main criterion for switching to Julia implementations. For instance, we should make sure that the Julia implementation is at least as accurate as the Rmath version, ideally on the whole input domain.

Reply:
Yeah #113 is complex enough that it's better to leave it alone. I'm just wondering what's the criterion to add some types to the list but not others. Maybe at least add small integer types?

Reply:
I added `IntX`s, `UIntX`s, and `Rational`s to the union as well since it would fix e.g. issues such as https://github.com/JuliaStats/Distributions.jl/pull/1387#issuecomment-913115634. I am not particularly happy about the hardcoded union and hence used `Real` instead whenever no Julia fallback existed.

Reply:
OK, thanks.  Some `Rational{Int}` values cannot be converted to `Float64` without a loss in precision, but maybe we don't care, just like `promote_type(Int64, Float64) == Float64` despite the precision loss for large values?

It would be nice to test the new types. :-)